frontend/src/scenes/billing/BillingProductAddon.tsx

```diff
@@ -174,7 +176,10 @@ export const BillingProductAddon = ({ addon }: { addon: BillingProductV2AddonTyp
                                             )
                                         }
                                     >
-                                        Add
+                                        {featureFlags[FEATURE_FLAGS.SUBSCRIBE_TO_ALL_PRODUCTS] &&
+                                        billing?.subscription_level === 'free'
+                                            ? 'Upgrade and add'
+                                            : 'Add'}
```

Comment:
That's a good point - it crossed my mind. We'd have to make sure that aligns with the UX from pay gates because right now it redirects to billing and scrolls to the addon.

Reply:
Made some changes by request of Cory to solve this for now. 
<img width="1149" alt="Screenshot 2024-06-18 at 12 21 52 PM" src="https://github.com/PostHog/posthog/assets/6392501/f155650b-d0c5-484c-a5cd-01ad7c114404">